python/pyspark/pandas/data_type_ops/num_ops.py

```diff
@@ -369,6 +376,9 @@ def floordiv(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
         _sanitize_list_like(right)
         if not is_valid_operand_for_numeric_arithmetic(right):
             raise TypeError("Floor division can not be applied to given types.")
+        spark_session = left._internal.spark_frame.sparkSession
+        use_try_divide = is_ansi_mode_enabled(spark_session)
+        safe_div = F.try_divide if use_try_divide else lambda x, y: x.__div__(y)  # type: ignore
```

Comment:
ditto.

Reply:
Sounds good, btw the code is adjusted to avoid E731